kitsune/src/http/pagination.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::Display;
+
+use axum::{
+    response::{IntoResponseParts, ResponseParts},
+    Json,
+};
+use http::{Error as HttpError, HeaderValue};
+
+use crate::error::Error;
+
+pub type PaginatedJsonResponse<T> = (Option<Link<Vec<(&'static str, String)>>>, Json<Vec<T>>);
+
+pub struct Link<T>(pub T);
+
+impl<T, K, V> IntoResponseParts for Link<T>
+where
+    T: IntoIterator<Item = (K, V)>,
+    K: Display,
+    V: Display,
+{
+    type Error = Error;
+
+    fn into_response_parts(self, mut res: ResponseParts) -> Result<ResponseParts, Self::Error> {
+        let value = self
+            .0
+            .into_iter()
+            .map(|(key, value)| format!("<{value}>; rel=\"{key}\""))
+            .collect::<Vec<String>>()
+            .join(", ");
+
+        res.headers_mut().insert(
+            "Link",
+            HeaderValue::from_str(&value).map_err(HttpError::from)?,
+        );
+
+        Ok(res)
+    }
+}
+
+pub fn new_link_header<I, D: Display, F: Fn(&I) -> D>(
```

Comment:
A silly idea I had.. what about making this an associated function of `Link` with the name `from_collection`.

And the parameter it takes is a struct annotated with the `TypedBuilder` macro to have the compile-time enforced builder pattern there.

Would probably provide a more readable code with less parameters

Reply:
Something like 

```rust
#[derive(TypedBuilder)]
struct CollectionParams {
    [fields here]
}
```

And the usage would be

```rust
let params = CollectionParams::builder()
    .base_uri(..)
    .build();

Link::from_collection(params)
```

Reply:
Builder patterns are nice, but I feel that here it's redundant. Maybe if we had some default arguments to explicitly provide or like 8 args making clippy scream at our code then yes, but not in this case.

Reply:
Yeah, fair. I still feel like making it an associated function of "Link" would be nice here though

Reply:
I agree 100%, sorry for not doing it before, but I made it into an associated fn now :)